Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -170,10 +173,49 @@ theorem primitive_element_inf_aux [IsSeparable F E] : ∃ γ : E, F⟮α, β⟯
   trans EuclideanDomain.gcd (?_ : E[X]) (?_ : E[X])
   · dsimp only
     convert (gcd_map (algebraMap F⟮γ⟯ E)).symm
-  · simp [map_comp, Polynomial.map_map, ← IsScalarTower.algebraMap_eq]
+  · simp only [map_comp, Polynomial.map_map, ← IsScalarTower.algebraMap_eq, Polynomial.map_sub,
+      map_C, AdjoinSimple.algebraMap_gen, map_add, Polynomial.map_mul, map_X]
     congr
 #align field.primitive_element_inf_aux Field.primitive_element_inf_aux
 
+/-- If `F` is infinite and `E/F` has only finitely many intermediate fields, then for any
+`α` and `β` in `E`, `F⟮α, β⟯` is generated by a single element. -/
+theorem primitive_element_inf_aux_of_finite_intermediateField
+    [Finite (IntermediateField F E)] : ∃ γ : E, F⟮α, β⟯ = F⟮γ⟯ := by
+  let f : F → IntermediateField F E := fun x => F⟮α + x • β⟯
+  obtain ⟨K, hinf⟩ := Finite.exists_infinite_fiber f
+  replace hinf : Nontrivial (Set.preimage f {K}) := inferInstance
+  obtain ⟨x, y, hneq⟩ := hinf.exists_pair_ne
+  replace hneq : (x : F) ≠ (y : F) := fun h => hneq (SetCoe.ext h)
+  have hx : f x = K := Set.mem_singleton_iff.1 $ Set.mem_preimage.1 $ Subtype.mem x
+  have hy : f y = K := Set.mem_singleton_iff.1 $ Set.mem_preimage.1 $ Subtype.mem y
+  simp only at hx
+  simp only at hy
+  use α + (x : F) • β
+  apply le_antisymm
+  · rw [hx, adjoin_le_iff]
+    have αxβ_in_K : α + (x : F) • β ∈ F⟮α + (x : F) • β⟯ := mem_adjoin_simple_self F _
+    have αyβ_in_K : α + (y : F) • β ∈ F⟮α + (y : F) • β⟯ := mem_adjoin_simple_self F _
+    rw [hx] at αxβ_in_K
+    rw [hy] at αyβ_in_K
+    have β_in_K := K.sub_mem αxβ_in_K αyβ_in_K
+    rw [show (α + (x : F) • β) - (α + (y : F) • β) = ((x : F) - (y : F)) • β by
+      rw [sub_smul]
+      abel1] at β_in_K
+    replace β_in_K := K.smul_mem β_in_K (x := ((x : F) - (y : F))⁻¹)
+    rw [smul_smul, inv_mul_eq_div, div_self (sub_ne_zero.2 hneq), one_smul] at β_in_K
+    have α_in_K : α ∈ K := by
+      rw [← add_sub_cancel α ((x : F) • β)]
+      exact K.sub_mem αxβ_in_K (K.smul_mem β_in_K (x := (x : F)))
+    exact fun x hx => by
```

Comment:
Can't this be `intro x hx` (or better yet `rintro x \<hx,\<hx\>\>`)

Reply:
It's copied from L121-L125. 😅 By the way, the `rintro` seems not working.

Reply:
Sorry I'll try and check my comments work next time ;) still I think something like that should be possible

Reply:
It works, but should be `rintro x (hx | hx)`. Thanks!